src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -46,21 +53,129 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /**
+   * Use playlist url to connect to YouTube API and set video ID.
+   *
+   * @param playlistUrl
+   */
+  private void setvideoId(String playlistUrl) {
+    String playlistId = getIdFromUrl(playlistUrl);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
     try {
-      ClientCredentials clientCredentials = clientCredentialsRequest.execute();
-      return clientCredentials.getAccessToken();
-    } catch (IOException | SpotifyWebApiException | ParseException e) {
-      System.out.println("Error: " + e.getMessage());
+      playlistItem = getPlaylistInfo(playlistId);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideos = parsePlaylistItem(playlistItemJson);
+    playlistSize = playlistVideos.size();
+    videoId = getRandomVideo(playlistVideos);
+  }
+
+  /**
+   * Parse Youtube URL for playlist ID.
+   *
+   * @param playlistUrl
+   * @return playlist ID
+   */
+  private String getIdFromUrl(String playlistUrl) {
+    if (playlistUrl.contains("youtube.com/playlist?list=")) {
+      int start = playlistUrl.indexOf("list=") + 5;
+      int end = playlistUrl.length();
+      if (playlistUrl.contains("&")) {
+        end = playlistUrl.indexOf("&", start);
+      }
+      return playlistUrl.substring(start, end);
+    } else {
+      System.err.println(playlistUrl + " is not a valid YouTube Playlist URL.");
+      return "";
+      // TODO @hdee: do something more complicated to handle this error.
+      // TODO @hdee: test this method.
     }
-    return "";
   }
 
+  /**
+   * Call function to create API service object.
+   *
+   * @throws GeneralSecurityException, IOException, GoogleJsonResponseException
+   */
+  public static PlaylistItemListResponse getPlaylistInfo(String playlistId)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    // Define and execute the API request
+    YouTube.PlaylistItems.List request =
+        youtubeService
+            .playlistItems()
+            .list("snippet")
+            .setMaxResults(MAX_RESULTS)
+            .setPlaylistId(playlistId);
+    PlaylistItemListResponse response = request.execute();
+    return response;
+  }
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * @return an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public static YouTube getService() throws GeneralSecurityException, IOException {
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    Credential credential = null;
```

Comment:
I think you can delete this line, and just use "null" as the third parameter in the line below. That would mean you can remove the "com.google.api.client.auth.oauth2.Credential" import and can remove the "com.google.oauth-client" dependency from the pom.xml file as well.